### PR TITLE
COMPAT: set ACLs with an empty AccessControlList

### LINKED ISSUE #304: COMPAT: set ACLs with an empty AccessControlList
COMPAT: set ACLs with an empty AccessControlList should not return any error.

```
<AccessControlPolicy xmlns= "http://s3.amazonaws.com/doc/2006-03-01/">
         <Owner>
                <ID>852b113e7a2f25102679df27bb0ae12b3f85be6BucketOwnerCanonicalUserID</ID>
                          <DisplayName>OwnerDisplayName</DisplayName>  
         </Owner>
         <AccessControlList></AccessControlList>
</AccessControlPolicy>
```

On **ceph/s3-tests**, these changes fix:

`ERROR: s3tests.functional.test_s3.test_bucket_acl_revoke_all`

### PR DESCRIPTION
**Pull request by nicolas2bert**
FIXES #304
I checked against AWS S3 and when:

- adding multiple AccessControlList section

OR

- AccessControlList section is not empty and there is no Grant section
it returns a MalformedACLError.

If AccessControlList section is empty `<AccessControlList></AccessControlList>`, AWS does not throw any error.

### CODE REVIEW

File: lib/utilities/aclUtils.js [hunk 1]
```diff
@@ -103,8 +103,8 @@ aclUtils.parseAclXml = function parseAclXml(toBeParsed, log, next) {
         }
         if (!result.AccessControlPolicy
                 || !result.AccessControlPolicy.AccessControlList
-                || !result.AccessControlPolicy
-                .AccessControlList[0].Grant) {
+                || (result.AccessControlPolicy.AccessControlList[0] !== '' &&
+                  !result.AccessControlPolicy.AccessControlList[0].Grant)) {
```
Comment (ttvi) [inline lib/utilities/aclUtils.js]: Are there multiple ACLs for a single bucket/object? In that case, what if the first ACL is OK but the rest is not?

File: tests/unit/api/bucketPutACL.js [hunk 2]
```diff
@@ -301,6 +301,104 @@ describe('putBucketACL API', () => {
...
+                assert.strictEqual(md.getAcl().READ.length, 0);
+                assert.strictEqual(md.getAcl().WRITE.length, 0);
+                assert.strictEqual(md.getAcl().WRITE_ACP.length, 0);
+                assert.strictEqual(md.getAcl().READ_ACP.length, 0);
+                done();
+            });
+        });
+    });
+
+    it('should set ACLs without AccessControlList section', done => {
```
Comment (ttvi) [inline tests/unit/api/bucketPutACL.js]: should not be able to set ACLs ...

File: tests/unit/api/bucketPutACL.js [hunk 3]
```diff
@@ -301,6 +301,104 @@ describe('putBucketACL API', () => {
...
+                    '<DisplayName>OwnerDisplayName</DisplayName>' +
+                  '</Owner>' +
+                  '<AccessControlList></AccessControlList>' +
+                '</AccessControlPolicy>',
+            url: '/?acl',
+            query: { acl: '' },
+        };
+
+        bucketPutACL(authInfo, testACLRequest, log, err => {
+            assert.strictEqual(err, undefined);
```
Comment (ttvi) [inline tests/unit/api/bucketPutACL.js]: I think you have mentioned that AWS doesn't accept empty or multiple ACLs but it doesn't seem to be what being tested here.

Comment (ttvi) [inline tests/unit/api/bucketPutACL.js]: unless "empty" means `'<AccessControlList></AccessControlList>'`. But even in such case, shouldn't it return the malformxml error here?

File: tests/unit/api/bucketPutACL.js (hunk 2 above)
Comment (nicolas2bert) [inline tests/unit/api/bucketPutACL.js]: Thanks

File: tests/unit/api/bucketPutACL.js (hunk 3 above)
Comment (nicolas2bert) [inline tests/unit/api/bucketPutACL.js]: My bad, I rephrased it to clarify it.

**Comment by nicolas2bert** [comment]
@ironman-machine please

**Pull request merged**